cmd/monitor/monitor_backup_schedule.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/fly-apps/postgres-flex/internal/flypg"
+)
+
+func monitorBackupSchedule(ctx context.Context, barman *flypg.Barman) {
+	// Determine when the last backup was taken.
+	lastBackupTime, err := barman.LastBackupTaken(ctx)
+	if err != nil {
+		log.Printf("Failed to resolve the last backup taken: %s", err)
+	}
+
+	// Ensure we have a least one backup before proceeding.
+	if lastBackupTime.IsZero() {
+		log.Println("No backups found! Performing the initial base backup.")
+
+		if err := performInitialBaseBackup(ctx, barman); err != nil {
+			log.Printf("Failed to perform the initial full backup: %s", err)
+			log.Printf("Backup scheduler will reattempt in 24 hours.")
+		}
+
+		lastBackupTime = time.Now()
+	}
+
+	// Calculate the time until the next backup is due.
+	timeUntilNextBackup := time.Until(lastBackupTime.Add(defaultFullBackupSchedule))
```

Comment:
Need to address this, doesn't currently consider a user specified `full_backup_frequency`.

Reply:
Fixed